Sources/Winget-AutoUpdate/Winget-Install.ps1

```diff
@@ -202,67 +301,85 @@ function Install-App ($AppID, $AppArgs) {
+        $retryCount = 0
+        if ($DisableRetry) {
+            $maxRetries = 0
+        }
+        else {
+            $maxRetries = 2
+        }
+        $uninstallSuccess = $false
+
+        while ($retryCount -le $maxRetries -and -not $uninstallSuccess) {
+            Write-ToLog "-> Uninstalling $AppID (Attempt: $($retryCount))..." "Yellow"
+            $retryCount++
+            $WingetArgs = "uninstall --id $AppID -e --accept-source-agreements -s winget" -split " "
+            Write-ToLog "-> Running: \"$Winget\" $WingetArgs"
+            & "$Winget" $WingetArgs | Where-Object { $_ -notlike "   *" } | Tee-Object -file $LogFile -Append
+            if (-not ([String]::IsNullOrEmpty($LASTEXITCODE))) {
+                $Script:ExitCode = $LASTEXITCODE
+            }
+            else {
+                $Script:ExitCode = 0
+            }
+
+            if ($exitCode -eq 0) {
+                $uninstallSuccess = $true
+                Write-ToLog "-> $AppID successfully uninstalled." "Green"
+            }
+            else {
+                Write-ToLog "-> $AppID uninstallation failed with Exit Code: $exitCode. Retrying... (Retry $retryCount of $maxRetries)" "Red"
+                Start-Sleep 5
+            }
+        }
 
-        if ($ModsUninstall) {
-            Write-ToLog "-> Modifications for $AppID during uninstall are being applied..." "Yellow"
-            & "$ModsUninstall"
+        if (-not $uninstallSuccess) {
+            Write-ToLog "-> $AppID uninstallation failed after $($maxRetries+1) attempts!" "Red"
         }
 
-        #Check if uninstall is ok
-        $IsInstalled = Confirm-Installation $AppID
-        if (!($IsInstalled)) {
-            Write-ToLog "-> $AppID successfully uninstalled." "Green"
+        # Apply post-uninstallation modifications if the uninstallation was successful
+        if ($uninstallSuccess) {
+            if ($ModsUninstall) {
+                Write-ToLog "-> Modifications for $AppID during uninstall are being applied..." "Yellow"
+                & "$ModsUninstall"
+            }
+
             if ($ModsUninstalled) {
                 Write-ToLog "-> Modifications for $AppID after uninstall are being applied..." "Yellow"
                 & "$ModsUninstalled"
             }
 
-            #Remove mods if deployed from Winget-Install
-            if (Test-Path ".\mods\$AppID-*") {
-                #Check if WAU default install path exists
-                $Mods = "$WAUModsLocation"
-                if (Test-Path "$Mods\$AppID*") {
-                    Write-ToLog "-> Remove $AppID modifications from WAU 'mods'"
-                    #Remove mods
-                    Remove-Item -Path "$Mods\$AppID-*" -Exclude "*uninstall*" -Force
-                }
-            }
-
-            #Remove from WAU White List if set
-            if ($WAUWhiteList) {
-                Remove-WAUWhiteList $AppID
+            # Remove leftover files and folders
+            $AppDataPath = "$env:LOCALAPPDATA\$AppID"
```

Comment:
$AppID is never the PATH and $LOCALAPPDATA only exist in user context...

Reply:
Thanks, I have removed that now